### ISSUE TITLE
Use the azure-account extension when working with Azure

### ISSUE THREAD

**Issue by itowlson** [Azure/vscode-kubernetes-tools#32]
Currently when we need to talk to Azure (for `Configure from Cluster` and `Create Cluster` when the user chooses an ACS or AKS cluster), we call the Azure CLI.  This requires the Azure CLI to be installed and logged in, and can affect global CLI state (active subscription) which might be different from the VS Code account state if the user is using the azure-account extension (https://marketplace.visualstudio.com/items?itemName=ms-vscode.azure-account).

We should consider using the azure-account extension and Node.js client instead of the CLI.  However, we need to consider the following issues:

* Extension dependency.  Currently we do not require users to install any Azure extensions, or the Azure CLI unless they are using Azure.  Many Kubernetes users will be using GKE or their own cluster and will _not_ want "this Azure crap" in their environments.  Therefore, we would need to either win over these users (tough because this can be an emotive issue) or move the cloud-specific touchpoints out into a separate extension (requires architectural effort and makes it harder for users to discover the additional extension they need for their cloud).

* Supported commands.  Azure CLI includes commands for retrieving `kubectl` binaries and credentials from the service.  It looks like the AKS REST API provides a way to retrieve credentials (what it calls 'access profiles'), though not as convenient as the CLI; the ACS REST API does not.  Worse, neither of these services appears to be surfaced in the current Node.js SDK.  So we incur some additional effort rewriting the CLI commands in Node.js.

**Comment by squillace** [comment]
Goal: People install the kubernetes extension and it works against any cluster for which there is a context you can detect/point to on disk.

We would like that to light up for specific cluster providers, such that -- for example -- installing a GKE account extension might enable seeing and/or creating/modifying clusters in GKE. Azure for Azure, and so on. 

As to the commands, at this point in time we want to support kubectl commands. That's it. But if there are specific commands that we'd expect to light up in provider extensions taking a dependency on this, then we might think about how to enable those.

**Comment by kyliel** [comment]
Agree, Ivan. Your proposal to use azure-account extension instead of the CLI is the right path for me. 

Extension dependency: A Kubernetes cluster could run on local or cloud (Azure, AWS, GKE...). This extension could be more generic but flexible to extend to cloud. If end user only wants to connect to a local cluster, no more extension is required after installing this one. If end user wants to connect to cloud, click "cloud" (we could discuss how to design this experience more) icon and select cloud provider to create or configure a cluster. Then end user will see the box to remind him/her installing cloud provider account extension (e.g. azure-account extension). 

Supported commands: I am asking the status of node.js SDK.

**Comment by amarzavery** [comment]
@itowlson @squillace @kyliel - The node sdk for AKS has been published. https://www.npmjs.com/package/azure-arm-containerservice. Let me know if you guys need anything else.
